### PR TITLE
Add config for VS2022

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
@wolfv @isuruf is this mergeable?

**Comment by isuruf** [comment]
You need to fix https://github.com/conda-forge/vc-feedstock/blob/main/recipe/activate.bat#L94

**Comment by mariusvniekerk** [comment]
> You need to fix [https://github.com/conda-forge/vc-feedstock/blob/main/recipe/activate.bat#L94](https://github.com/conda-forge/vc-feedstock/blob/main/recipe/activate.bat?rgh-link-date=2022-06-16T19%3A54%3A31Z#L94)

Doesn't that https://github.com/conda-forge/vc-feedstock/blob/main/recipe/activate.bat#L80

**Comment by isuruf** [comment]
No, that's when you have have the visual studio version and the vc version matching. What I'm talking about is when they don't, but you can still target the vc version requested by the user even with a newer visual studio version.

**Comment by mariusvniekerk** [comment]
@isuruf does this look okay now?

File: recipe/activate.bat [hunk 1]
```diff
@@ -91,8 +91,13 @@ IF @{year} GEQ 2019  (
 echo "NEWER_VS_WITH_OLDER_VC=%NEWER_VS_WITH_OLDER_VC%"
 
 IF "%NEWER_VS_WITH_OLDER_VC%" == "1" (
-    set "CMAKE_GEN=Visual Studio 16 2019"
-    set "USE_NEW_CMAKE_GEN_SYNTAX=1"
+    IF @{year} GEQ 2022 (
```
Comment (isuruf) [inline recipe/activate.bat:94]: This is not correct. year has to be deduced from `VSINSTALLDIR` which can have `2022` for `@{year}=2019` or `@{year}=2017`

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
? which additional binaries are you talking about?

**Comment by h-vetinari** [comment]
> ? which additional binaries are you talking about?

MS doesn't touch `vcruntime140.dll` for forwards-compatibility reason of old code, that's why VS2019 had some pure additions, e.g. `vcruntime140_1.dll` / `msvcp140_atomic_wait.dll` / `msvcp140_codecvt_ids.dll`.

I thought it was quite likely that VS2022 would have some more additional DLLs for the same reason, but if I read the logs right, no new DLLs were added after all, i.e.
```
Found DLL: concrt140.dll
Found DLL: msvcp140.dll
Found DLL: msvcp140_1.dll
Found DLL: msvcp140_2.dll
Found DLL: msvcp140_atomic_wait.dll
Found DLL: msvcp140_codecvt_ids.dll
Found DLL: vcamp140.dll
Found DLL: vccorlib140.dll
Found DLL: vcomp140.dll
Found DLL: vcruntime140.dll
Found DLL: vcruntime140_1.dll
```
should all be covered by the current tests.

**Pull request merged**